Gems/Atom/Tools/AtomToolsFramework/Code/Source/AssetBrowser/AtomToolsAssetBrowser.cpp

```diff
@@ -73,38 +60,82 @@ namespace MaterialEditor
+        connect(m_ui->m_assetBrowserTreeViewWidget, &AssetBrowserTreeView::activated, this, &AtomToolsAssetBrowser::OpenSelectedEntries);
+        connect(m_ui->m_assetBrowserTreeViewWidget, &AssetBrowserTreeView::selectionChangedSignal, this, &AtomToolsAssetBrowser::UpdatePreview);
+        connect(m_ui->m_viewOptionButton, &QPushButton::clicked, this, &AtomToolsAssetBrowser::OpenOptionsMenu);
+        connect(
+            m_ui->m_searchWidget->GetFilter().data(), &AssetBrowserEntryFilter::updatedSignal, m_filterModel,
+            &AssetBrowserFilterModel::filterUpdatedSlot);
+    }
+
+    AtomToolsAssetBrowser::~AtomToolsAssetBrowser()
+    {
+        // Maintains the tree expansion state between runs
+        m_ui->m_assetBrowserTreeViewWidget->SaveState();
+        AZ::TickBus::Handler::BusDisconnect();
+    }
+
+    void AtomToolsAssetBrowser::SetFilterState(const AZStd::string& category, const AZStd::string& displayName, bool enabled)
+    {
+        m_ui->m_searchWidget->SetFilterState(category.c_str(), displayName.c_str(), enabled);
+    }
+
+    void AtomToolsAssetBrowser::SetOpenHandler(AZStd::function<void(const AZStd::string&)> openHandler)
+    {
+        m_openHandler = openHandler;
+    }
+
+    void AtomToolsAssetBrowser::SelectEntries(const AZStd::string& absolutePath)
+    {
+        if (!absolutePath.empty())
         {
-            const bool hasFilter = !m_ui->m_searchWidget->GetFilterString().isEmpty();
-            constexpr bool selectFirstFilteredIndex = true;
-            m_ui->m_assetBrowserTreeViewWidget->UpdateAfterFilter(hasFilter, selectFirstFilteredIndex);
-        });
-        connect(m_ui->m_assetBrowserTreeViewWidget, &AssetBrowserTreeView::activated, this, &MaterialBrowserWidget::OpenSelectedEntries);
-        connect(m_ui->m_assetBrowserTreeViewWidget, &AssetBrowserTreeView::selectionChangedSignal, [this]() {
-            const auto& selectedAssets = m_ui->m_assetBrowserTreeViewWidget->GetSelectedAssets();
-            if (!selectedAssets.empty())
+            // Selecting a new asset in the browser is not guaranteed to happen immediately.
+            // The asset browser model notifications are sent before the model is updated.
+            // Instead of relying on the notifications, queue the selection and process it on tick until this change occurs.
+            m_pathToSelect = absolutePath;
+            AzFramework::StringFunc::Path::Normalize(m_pathToSelect);
+            AZ::TickBus::Handler::BusConnect();
+        }
+    }
+
+    void AtomToolsAssetBrowser::OpenSelectedEntries()
+    {
+        const AZStd::vector<AssetBrowserEntry*> entries = m_ui->m_assetBrowserTreeViewWidget->GetSelectedAssets();
+
+        const int multiSelectPromptThreshold = 10;
+        if (entries.size() >= multiSelectPromptThreshold)
+        {
+            QMessageBox::StandardButton result = QMessageBox::question(
+                QApplication::activeWindow(),
```

Comment:
Nit: Please use `AzToolsFramework::GetActiveWindow()` instead, it is a safer helper in case the `QApplication` version returns nullptr